client/src/routes/temperaturesensor/TemperatureWidget/TemperatureWidget.js

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import withStyles from 'isomorphic-style-loader/withStyles';
+import Switch from 'react-switch';
+import { Spinner, Alert, Row, Col } from 'react-bootstrap';
+import { connect } from 'react-redux';
+import style from './TemperatureWidget.css';
+import {
+  getTemperatureSensorsData,
+  changeTemperatureStatus,
+} from '../../../actions/temperature';
+
+const styleSpiner = {
+  marginLeft: '0.5em',
+};
+
+class TemperatureWidget extends Component {
+  static propTypes = {
+    dispatchChangeStatus: PropTypes.func.isRequired,
+    dispatchWidgetData: PropTypes.func.isRequired,
+    error: PropTypes.string.isRequired,
+    events: PropTypes.arrayOf(
+      PropTypes.shape({
+        Created: PropTypes.string.isRequired,
+        CreatedAt: PropTypes.string.isRequired,
+        EventID: PropTypes.number.isRequired,
+        ID: PropTypes.number.isRequired,
+        UpdatedAt: PropTypes.string.isRequired,
+        degree: PropTypes.number.isRequired,
+        device_type: PropTypes.string.isRequired,
+        name: PropTypes.string.isRequired,
+      }),
+    ).isRequired,
+    info: PropTypes.shape({
+      Name: PropTypes.string.isRequired,
+      Status: PropTypes.number.isRequired,
+      Type: PropTypes.string.isRequired,
+    }).isRequired,
+  };
+
+  handleOnClick = status => {
+    const { dispatchChangeStatus } = this.props;
+    if (status === 1) {
+      dispatchChangeStatus(false);
+    } else {
+      dispatchChangeStatus(true);
+    }
+  };
+
+  render() {
+    const { info, dispatchWidgetData, events, error } = this.props;
+
+    let degree;
+    let createdAt;
+    if (events.length !== 0) {
+      degree = events.slice(-1)[0].degree;
+      createdAt = moment(events.slice(-1)[0].CreatedAt).format(
+        'DD-MM HH:mm:ss',
+      );
+    }
+
+    dispatchWidgetData({
```

Comment:
why do you dispatch it in render? Do you understand that dispatching an action can cause an additional render? Consider moving this call to constructor or `componentDidMount` (or `componentDidUpdate` if you are crazy)